net-fs/samba/patches/samba-4.5.5.patchset

```diff
@@ -0,0 +1,82 @@
+From 942e8a9a2ca249e9121181217178cb3a3ff6fc19 Mon Sep 17 00:00:00 2001
+From: Sergei Reznikov <diver@gelios.net>
+Date: Fri, 11 Apr 2014 15:46:09 +0400
+Subject: applying patch samba-3.6.23.patch
+
+
+diff --git a/source3/utils/net_time.c b/source3/utils/net_time.c
+index 0edb58c..2b8d365 100644
+--- a/source3/utils/net_time.c
++++ b/source3/utils/net_time.c
+@@ -105,6 +105,14 @@ int net_time_usage(struct net_context *c, int argc, const char **argv)
+ 	return -1;
+ }
+ 
++#ifdef __HAIKU__
++int settimeofday(struct timeval *tv, struct timezone *tz)
++{
++set_real_time_clock(tv->tv_sec);
++return(0);
++}
++#endif //__HAIKU__
++
+ /* try to set the system clock */
+ static int net_time_set(struct net_context *c, int argc, const char **argv)
+ {
+-- 
+1.8.3.4
+
+From 4bfe334dc05a0f8ba474209c14d0f9ac16b720ca Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Wed, 1 Mar 2017 00:02:17 +0100
+Subject: [PATCH] Haiku build fix
+
+---
+ lib/replace/system/time.h | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/lib/replace/system/time.h b/lib/replace/system/time.h
+index b6d2609..d02c237 100644
+--- a/lib/replace/system/time.h
++++ b/lib/replace/system/time.h
+@@ -80,7 +80,7 @@ typedef int clockid_t;
+ int rep_clock_gettime(clockid_t clk_id, struct timespec *tp);
+ #endif
+ /* make sure we have a best effort CUSTOM_CLOCK_MONOTONIC we can rely on */
+-#if defined(CLOCK_MONOTONIC)
++#if defined(CLOCK_MONOTONIC) && !defined(__HAIKU__)
+ #define CUSTOM_CLOCK_MONOTONIC CLOCK_MONOTONIC
+ #elif defined(CLOCK_HIGHRES)
+ #define CUSTOM_CLOCK_MONOTONIC CLOCK_HIGHRES
+-- 
+2.11.0
+
+From a8da2902b4d47841317542c0169afc47515a9f59 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Wed, 1 Mar 2017 08:30:34 +0100
+Subject: [PATCH] PIPE_BUF paatch
+
+---
+ lib/tevent/testsuite.c | 7 +++++++
+ 1 file changed, 7 insertions(+)
+
+diff --git a/lib/tevent/testsuite.c b/lib/tevent/testsuite.c
+index b37c7b1..ac7e9c8 100644
+--- a/lib/tevent/testsuite.c
++++ b/lib/tevent/testsuite.c
+@@ -36,6 +36,13 @@
+ #include <assert.h>
+ #endif
+ 
++#ifdef __HAIKU__
++#include <posix/limits.h>
++#ifndef PIPE_BUF
++#define PIPE_BUF _POSIX_PIPE_BUF
```

Comment:
It seems Haiku defined this value here https://github.com/haiku/haiku/blob/master/headers/private/system/vfs_defs.h#L29
Probably needs to be patched in Haiku